authentik/providers/oauth2/views/token_revoke.py

```diff
@@ -27,7 +31,9 @@ def from_request(request: HttpRequest) -> "TokenRevocationParams":
         """Extract required Parameters from HTTP Request"""
         raw_token = request.POST.get("token")
 
-        provider = authenticate_provider(request)
+        provider, _, _ = provider_from_request(request)
+        if provider and provider.client_type == ClientTypes.CONFIDENTIAL:
+            provider = authenticate_provider(request)
```

Comment:
We may want to further split authenticate_provider to avoid parsing the request multiple times.